caput/pipeline/tasklib/base.py

```diff
@@ -199,31 +237,31 @@ def __init__(self):
 
 
 class ContainerTask(MPILoggedTask, extensions.BasicContMixin):
-    """Process a task with at most one input and output.
+    """Implements a task whose inputs and outputs are :py:class:`~caput.memdata.BasicCont` objects.
 
-    Both input and output are expected to be :class:`memh5.BasicCont` objects.
-    This class allows writing of the output when requested.
+    This task implements writing of the output when requested, and handles various
+    types of metadata associated with the container objects.
 
-    Tasks inheriting from this class should override `process` and optionally
-    :meth:`setup` or :meth:`finish`. They should not override :meth:`next`.
+    Tasks inheriting from this class should override :py:meth:`~.ContainerTask.process`
+    and optionally :py:meth:`~.ContainerTask.setup` or :py:meth:`~.ContainerTask.finish`.
+    They should not override :py:meth:`~.ContainerTask.next`.
 
-    If the value of :attr:`input_root` is anything other than the string "None"
-    then the input will be read (using :meth:`read_input`) from the file
-    ``self.input_root + self.input_filename``.  If the input is specified both as
+    If the value of :py:attr:`~.ContainerTask.input_root` is anything other than the string "None"
+    then the input will be read (using :py:meth:`~.ContainerTask.read_input`) from the file
+    `self.input_root + self.input_filename`.  If the input is specified both as
     a filename and as a product key in the pipeline configuration, an error
     will be raised upon initialization.
 
-
-    If the value of :attr:`output_root` is anything other than the string
-    "None" then the output will be written (using :meth:`write_output`) to the
+    If the value of :py:attr:`~.ContainerTask.output_root` is anything other than the string
+    ``None`` then the output will be written (using :meth:`write_output`) to the
```

Comment:
Given `output_root` is deprecated, it's probably best to de-emphasise this by moving this sentence down in the parameter description for `output_root`.